entrypoint/entrypoint.go

```diff
@@ -2,14 +2,25 @@ package entrypoint
 
 import (
 	"os"
+
 	"github.com/urfave/cli"
+
 	"github.com/gruntwork-io/gruntwork-cli/errors"
 	"github.com/gruntwork-io/gruntwork-cli/logging"
 )
 
 const defaultSuccessExitCode = 0
 const defaultErrorExitCode = 1
 
+// Wrapper around cli.NewApp that sets the help text printer.
+func NewApp() *cli.App {
+	cli.HelpPrinter = WrappedHelpPrinter
+	cli.AppHelpTemplate = CLI_APP_HELP_TEMPLATE
+	cli.CommandHelpTemplate = CLI_COMMAND_HELP_TEMPLATE
```

Comment:
That should work. Actually, these are global variables in `urfave/cli` that are read in as part of the `help` command, so a later overwrites will take precedence. You just have to override after the `NewApp` call, as opposed to before. That said, it doesn't handle the elastic tabstop output table so for that example the output is not so nice (pasted below). That said, once I switched the template to raw tabs, it looked better so may not be worth generalizing.
#### No tabs:
```
DESCRIPTION:
   terragrunt - Terragrunt is a thin wrapper for Terraform that provides extra
   tools for working with multiple
   Terraform modules, remote state, and locking. For documentation, see
   https://github.com/gruntwork-io/terragrunt/.

USAGE:
   terragrunt <COMMAND>

COMMANDS:
   plan-all             Display the plans of a 'stack' by running 'terragrunt
   plan' in each subfolder
   apply-all            Apply a 'stack' by running 'terragrunt apply' in each
   subfolder
   output-all           Display the outputs of a 'stack' by running 'terragrunt
   output' in each subfolder
   destroy-all          Destroy a 'stack' by running 'terragrunt destroy' in
   each subfolder
   validate-all         Validate 'stack' by running 'terragrunt validate' in
   each subfolder
   *                    Terragrunt forwards all other commands directly to
   Terraform

GLOBAL OPTIONS:
   terragrunt-config                    Path to the Terragrunt config file.
   Default is terraform.tfvars.
   terragrunt-tfpath                    Path to the Terraform binary. Default is
   terraform (on PATH).
   terragrunt-no-auto-init              Don't automatically run 'terraform init'
   during other terragrunt commands. You must run 'terragrunt init' manually.
   terragrunt-no-auto-retry             Don't automatically re-run command in
   case of transient errors.
   terragrunt-non-interactive           Assume "yes" for all prompts.
   terragrunt-working-dir               The path to the Terraform templates.
   Default is current directory.
   terragrunt-download-dir              The path where to download Terraform
   code. Default is .terragrunt-cache in the working directory.
   terragrunt-source                    Download Terraform configurations from
   the specified source into a temporary folder, and run Terraform in that
   temporary folder.
   terragrunt-source-update             Delete the contents of the temporary
   folder to clear out any old, cached source code before downloading new source
   code into it.
   terragrunt-iam-role                 Assume the specified IAM role
                                       before executing Terraform.
                                       Can also be set via the
                                       TERRAGRUNT_IAM_ROLE
                                       environment variable.
   terragrunt-ignore-dependency-errors  *-all commands continue processing
   components even if a dependency fails.
   terragrunt-exclude-dir               Unix-style glob of directories to
   exclude when running *-all commands

VERSION:


AUTHOR(S):
   Gruntwork <www.gruntwork.io>
```

#### with tabs
```
DESCRIPTION:                                                                                                                                         [65/7627]
   terragrunt - Terragrunt is a thin wrapper for Terraform that provides extra
   tools for working with multiple
   Terraform modules, remote state, and locking. For documentation, see
   https://github.com/gruntwork-io/terragrunt/.

USAGE:
   terragrunt <COMMAND>

COMMANDS:
   plan-all      Display the plans of a 'stack' by running 'terragrunt plan'
                 in each subfolder
   apply-all     Apply a 'stack' by running 'terragrunt apply' in each
                 subfolder
   output-all    Display the outputs of a 'stack' by running 'terragrunt
                 output' in each subfolder
   destroy-all   Destroy a 'stack' by running 'terragrunt destroy' in each
                 subfolder
   validate-all  Validate 'stack' by running 'terragrunt validate' in each
                 subfolder
   *             Terragrunt forwards all other commands directly to Terraform

GLOBAL OPTIONS:
   terragrunt-config                    Path to the Terragrunt config file. Default is
                                        terraform.tfvars.
   terragrunt-tfpath                    Path to the Terraform binary. Default is terraform
                                        (on PATH).
   terragrunt-no-auto-init              Don't automatically run 'terraform init'
                                        during other terragrunt commands. You must run
                                        'terragrunt init' manually.
   terragrunt-no-auto-retry             Don't automatically re-run command in case of
                                        transient errors.
   terragrunt-non-interactive           Assume "yes" for all prompts.
   terragrunt-working-dir               The path to the Terraform templates. Default is
                                        current directory.
   terragrunt-download-dir              The path where to download Terraform code.
                                        Default is .terragrunt-cache in the working
                                        directory.
   terragrunt-source                    Download Terraform configurations from the specified
                                        source into a temporary folder, and run Terraform in
                                        that temporary folder.
   terragrunt-source-update             Delete the contents of the temporary folder
                                        to clear out any old, cached source code
                                        before downloading new source code into it.
   terragrunt-iam-role                  Assume the specified IAM role before executing
                                        Terraform. Can also be set via the
                                        TERRAGRUNT_IAM_ROLE environment variable.
   terragrunt-ignore-dependency-errors  *-all commands continue processing
                                        components even if a dependency
                                        fails.
   terragrunt-exclude-dir               Unix-style glob of directories to exclude when
                                        running *-all commands

VERSION:


AUTHOR(S):
   Gruntwork <www.gruntwork.io>
```